precompiles/parachain-staking/src/lib.rs

```diff
@@ -88,6 +88,23 @@ where
 		Ok(points)
 	}
 
+	#[precompile::public("awardedPoints(uint256,address)")]
+	#[precompile::view]
+	fn awarded_points(
+		handle: &mut impl PrecompileHandle,
+		round: SolidityConvert<U256, u32>,
+		candidate: Address,
+	) -> EvmResult<u32> {
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+
+		let round = round.converted();
+		let candidate = Runtime::AddressMapping::into_account_id(candidate.0);
+
+		let points = <pallet_parachain_staking::Pallet<Runtime>>::awarded_pts(&round, &candidate);
```

Comment:
One problem with this is that it will return `0` once the `AwardedPts` storage for the given round/candidate have been paid out.

This is especially cumbersome since we no longer know (from storage) whether there never was points for that candidate or they have already been paid out.

We could simply document this, but we could probably also return some kind of result indicating that it may have been removed.

Reply:
Yeah unfortunately there's no way to know this in the current context, so adding a doc comment seems the best choice.

Reply:
Yes, probably best to document it. We *could* query the current round and payout delay to distinguish between:

* result was paid out already iff it existed in the first place
* result should be in pending-payout stage (but still valid only for a selected collator)
* result could not exist yet (in the future) (?)

I'm leaning towards just documenting it :)

One further issue is that you could submit a txn which wants to query this value and it would behave differently depending on how long it sits in the txpool, which makes me wonder whether it's useful at all in that scenario...

Reply:
Yeah I guess this is probably a best-effort scenario.